terraform/README.md

```diff
@@ -0,0 +1,73 @@
+# SD-Core NMS K8s Terraform Module
+
+This SD-Core NMS K8s Terraform module aims to deploy the [sdcore-nms-k8s charm](https://charmhub.io/sdcore-nms-k8s) via Terraform.
+
+## Getting Started
+
+### Prerequisites
+
+The following software and tools needs to be installed and should be running in the local environment.
+
+- `microk8s`
+- `juju 3.x`
+- `terrafom`
+
+### Deploy the sdcore-nms-k8s charm using Terraform
+
+Make sure that `storage`, `multus` and `metallb` plugins are enabled for Microk8s:
+
+```console
+sudo microk8s enable hostpath-storage multus
+sudo microk8s enable metallb:10.0.0.2-10.0.0.4
+```
+
+Add a Juju model:
+
+```console
+juju add model <model-name>
+```
+
+Initialise the provider:
+
+```console
+terraform init
+```
+
+Customize the configuration inputs under `terraform.tfvars` file according to requirement.
+
+Replace the values in the `terraform.tfvars` file:
+
+```yaml
+# Mandatory Config Options
+model_name               = "put your model-name here"
+traefik_application_name = "put your nrf app name here"
+webui_application_name   = "put your nrf app name here"
+```
+
+Run Terraform Plan by providing var-file:
+
+```console
+terraform plan -var-file="terraform.tfvars" 
+```
+
+Deploy the resources, skip the approval:
```

Comment:
```suggestion
Deploy the resources:
```

Reply:
I think it's not bad to say `Deploy the resources by skipping the approval`

Reply:
I was mostly concerned about the `by` (which I see is gone now). `By` suggests that it's skipping the approval that deploys the resources, which is not true. Issuing the `terraform apply` command deploys the resources. Skipping the approval is just an optional argument. That's why I suggested to remove this part completely.

Reply:
`skip approval` is removed.